launch.md

```diff
@@ -0,0 +1,150 @@
+
+**externalConsole**: If set to `true`, launches an external console for the application. If `false`, no console is launched and VS Code's debugging console is used. Note this option is ignored in some cases for technical reasons.
+                            
+**logging**: Optional flags to determine what types of messages should be logged to the Debug Console.
+  * **exceptions**: Optional flag to determine whether exception messages should be logged to the Debug Console. Defaults to true.
+  * **moduleLoad**: Optional flag to determine whether module load events should be logged to the Debug Console. Defaults to true.
+  * **programOutput**: Optional flag to determine whether program output should be logged to the Debug Console. Defaults to true.
+  * **engineLogging**: Optional flag to determine whether diagnostic engine logs should be logged to the Debug Console. Defaults to false.
+  * **trace**: Optional flag to determine whether diagnostic adapter command tracing should be logged to the Debug Console. Defaults to false.
+  * **traceResponse**: Optional flag to determine whether diagnostic adapter command and response tracing should be logged to the Debug Console. Defaults to false.
+
+**visualizerFile**: .natvis file to be used when debugging. See [Natvis syntax reference](http://aka.ms/natvis#Anchor_8) for information on how to create natvis files.
+
+**Example:**
+
+```json
+{
+   "name": "C++ Launch (Windows)",
+   "type": "cppvsdbg",
+   "request": "launch",
+   "program": "C:\\app1\\Debug\\app1.exe",
+   "symbolSearchPath": "C:\\Symbols;C:\\SymbolDir2",
+   "externalConsole": true,
+   "logging": {
+       "moduleLoad": false,
+       "trace": true
+    },
+   "visualizerFile": "${workspaceRoot}/my.natvis"
+}
+```
+
+
+# Configure the target application
+
+The following options enable you to modify the state of the target application when it is launched
+
+**args**: JSON array of command line arguments to pass to the program when it is launched. Example `["arg1", "arg2]`.
+
+**cwd**: Sets the the working directory of the application launched by the debugger.
+
+**environment**: Environment variables to add to the environment for the program. Example: `[ { "name": "squid", "value": "clam" } ]`.
+
+**Example:**
+
+```json
+{
+   "name": "C++ Launch",
+   "type": "cppdbg",
+   "request": "launch",
+   "program": "${workspaceRoot}/a.out",
+   "args": ["arg1", "arg2"],
+   "environment": [{"name": "squid", "value": "clam"}],
+   "cwd": "${workspaceRoot}"
+}
+```
+
+## Customizing GDB or LLDB 
+
+You can change the behavior of GDB or LLDB by setting the following options.
+
+**MIMode**: Indicates the debugger that VS Code will connect to. Must be set to `gdb` or `lldb`.  This is pre-configured on a per-operating system basis and can be changed as needed.
+
+**miDebuggerPath**: The path to the debugger (such as gdb). When unspecified, it will search the operating system's PATH variable first for the debugger. default": "/usr/bin/gdb".
```

Comment:
since we search the path, the default really isn't /usr/bin/gdb any more. I wonder if we should just remove the default on this one? In the schema, this is actually the generated value if the user decides to add the field.